lib/datadog.ts

```diff
@@ -62,4 +72,30 @@ export class Datadog extends cdk.Construct {
       this.transport.setEnvVars(lambdaFunctions);
     }
   }
+
+  public validateProps(props: DatadogProps) {
+    const siteList: string[] = ["datadoghq.com", "datadoghq.eu", "us3.datadoghq.com", "ddog-gov.com"];
+
+    if (props.extensionLayerVersion !== undefined) {
+      if (props.forwarderARN !== undefined) {
+        throw new Error("`extensionLayerVersion` and `forwarderArn` cannot be set at the same time.");
+      }
+      if (props.apiKey === undefined && props.apiKMSKey === undefined) {
+        throw new Error("When `extensionLayer` is set, `apiKey` or `apiKMSKey` must also be set.");
+      }
+    }
+    if (props.site !== undefined && !siteList.includes(props.site.toLowerCase())) {
+      console.log(
+        "Warning: Invalid site URL. Must be either datadoghq.com, datadoghq.eu, us3.datadoghq.com, or ddog-gov.com.",
+      );
+    }
+    if (
+      (props.apiKey !== undefined && props.apiKMSKey !== undefined && props.flushMetricsToLogs === false) ||
+      (props.apiKey === undefined && props.apiKMSKey === undefined && props.flushMetricsToLogs === false)
+    ) {
+      throw new Error(
```

Comment:
I know this was already here, but it seems like this is the wrong error message for `(props.apiKey === undefined && props.apiKMSKey === undefined && props.flushMetricsToLogs === false)`. I.e. when neither is set, why do we tell the user that they are mutually exclusive? Maybe it would be better to say: "When `flushMetricsToLogs` is false, `apiKey` or `apiKMSKey` must also be set." in that situation. Kind of like we do for when there is an extension layer version.

Reply:
In other words, I think these two situations deserve two different error messages.

Reply:
Yeah I thought the way it was phrased before was odd